src/System.Private.Xml/src/System/Xml/XmlDownloadManager.cs

```diff
@@ -36,22 +36,196 @@ internal partial class XmlDownloadManager
             RequestCachePolicy cachePolicy)
         {
-            HttpClient client = new HttpClient();
-            HttpRequestMessage req = new HttpRequestMessage(HttpMethod.Get, uri);
-
-            lock (this)
+            WebRequest req = WebRequest.Create(uri);
+            if (credentials != null)
+            {
+                req.Credentials = credentials;
+            }
+            if (proxy != null)
             {
-                if (_connections == null)
+                req.Proxy = proxy;
+            }
+            if (cachePolicy != null)
+            {
+                req.CachePolicy = cachePolicy;
+            }
+            WebResponse resp = req.GetResponse();
+            HttpWebRequest webReq = req as HttpWebRequest;
+            if (webReq != null)
+            {
+                lock (this)
                 {
-                    _connections = new Hashtable();
+                    if (_connections == null)
+                    {
+                        _connections = new Hashtable();
+                    }
+                    OpenedHost openedHost = (OpenedHost)_connections[webReq.Address.Host];
+                    if (openedHost == null)
+                    {
+                        openedHost = new OpenedHost();
+                    }
+
+                    if (openedHost.nonCachedConnectionsCount < webReq.ServicePoint.ConnectionLimit - 1)
+                    {
+                        // we are not close to connection limit -> don't cache the stream
+                        if (openedHost.nonCachedConnectionsCount == 0)
+                        {
+                            _connections.Add(webReq.Address.Host, openedHost);
+                        }
+                        openedHost.nonCachedConnectionsCount++;
+                        return new XmlRegisteredNonCachedStream(resp.GetResponseStream(), this, webReq.Address.Host);
+                    }
+                    else
+                    {
+                        // cache the stream and save the connection for the next request
+                        return new XmlCachedStream(resp.ResponseUri, resp.GetResponseStream());
+                    }
                 }
             }
+            else
+            {
+                return resp.GetResponseStream();
+            }
+        }
+    }
+
+    //
+    // OpenedHost
+    //
+    internal class OpenedHost
+    {
+        internal int nonCachedConnectionsCount;
+    }
+
+    //
+    // XmlRegisteredNonCachedStream
+    //
+    internal class XmlRegisteredNonCachedStream : Stream
+    {
+        protected Stream stream;
+        private XmlDownloadManager _downloadManager;
+        private string _host;
+
+        internal XmlRegisteredNonCachedStream(Stream stream, XmlDownloadManager downloadManager, string host)
+        {
+            this.stream = stream;
+            _downloadManager = downloadManager;
+            _host = host;
+        }
+
```

Comment:
The desktop one has Dispose (and a Finalizer) did you mean to leave these out?
http://referencesource.microsoft.com/#System.Xml/System/Xml/XmlDownloadManager.cs,113